### ISSUE TITLE
Errors

### ISSUE THREAD

**Issue by Tyloonie** [JCulley/ProQuests-Documentation#104]
Spigot Account: https://www.spigotmc.org/members/tyloonie.130061/

I'm using the latest version of **ProQuests**, Spigot version CraftBukkit version git-Spigot-b6ecf3b-fb423b8 (MC: 1.12.2) (Implementing API version 1.12.2-R0.1-SNAPSHOT). **ProVotes** version 1.3.5.4.

There are no specific logs to apply to my ticket, so I can only explain it as good as possible.

Referring my review from yesterday, I still think it's an awesome plugin. I do however have a few issues to address. 

The first issue is with the permission **proquests.quest.autostart.*** which does not function as expected. On your first login, it only autostarts a few of the quests. After the plugin has registered your data however and the server has been restarted, all quests are now automatically active on your first login after the server restart. Could this be fixed so that at first join, all quests will be automatically started if the player has the **proquests.quest.autostart.*** permission?

The second issue is with saving Quest NPCs. It seems like the plugin does not save the data of newly created Quest NPCs on /proquest reload or on server restart. It's therefore impossible for me to create any NPCs related to ProQuests.

The third issue is with ProVotes. After completing a quest where you have to vote once, the quest sometimes still gives you rewards on voting even after you completed that quest. 

Think that's it. I've attached a screenshot of the GUI when not all quests automatically restart to give you an idea of how it looks.

Thanks!
![30184521_1830204390337549_1562175942_n](https://user-images.githubusercontent.com/19597444/38977932-1bec853e-43b6-11e8-92f1-16cdec5fb47c.png)

**Comment by JCulley** [comment]
Please provide your log file from when the 1st and 2nd errors occurred. The 3rd is not an error as the quest has no intention of canceling any votes made so it will always give you any voting rewards if eligible and applicable.

**Comment by Tyloonie** [comment]
Well, there are no stacktraces or exceptions if that's what you mean. It simply just doesn't work. There is no way to really trace this. I'll take an extra look to see if I can find any errors.

In terms of the 3rd error. I don't have any intend for the plugin to cancel rewards from voting. I just want the plugin to cancel rewards from a quest which has already been completed. In this case, if you vote once and complete the quest where you have to vote once, you will still sometimes get the reward from the quest - even if you've already completed it.

**Comment by JCulley** [comment]
Can you please provide what I have requested, I personally would like to look at it for me to continue to look into your issue. I misread the “error” as you put it but in programming things don’t happen “sometimes” there is always certain circumstances causing the occurance so please also provide me your config files and quest files.

Please do not question my request of these documents as overall it will make it easier to help you without this back and forth.

Thanks!

**Comment by Tyloonie** [comment]
Yes Sir!

ProQuests startup log: https://hastebin.com/jeqaperezu.md

config.yml and my 3 vote quests:
[ProQuests.zip](https://github.com/JCulley/ProQuests-Documentation/files/1928308/ProQuests.zip)

**Comment by Tyloonie** [comment]
Hello, did any of the information I provided help?

Let me know if you need anything else.

**Comment by Tyloonie** [comment]
Hello, 10 days later -
Just gonna bump this quick. You've made any progress man?
Thanks!

**Comment by JCulley** [comment]
Haven't had any time to look into it, when I get a chance I will.  There are a couple issues currently open I need to investigate before I get back to you.

**Comment by Tyloonie** [comment]
Okay, fair. Let me know when you've made any progress.

**Comment by JCulley** [comment]
1) They need to have the proquests.quest.autostart permission otherwise none will start automatically. Otherwise from a technical standpoint it will start every quest a player is eligible to begin the second the login.

2) Tested creating a new NPC and it saved to the folder perfectly fine, so not sure what your issue is

3) There was something I noticed in the code related to marking a quest as completed in general which I have addressed however there were numerous safe checks in place and based on those the error you're explaining wouldn't have been possible from a technical standpoint.

As all of your errors have been addressed I am closing this issue. Be sure to update to the new version.
